### ISSUE TITLE
New Room List: Configuration of room list not persisted across application launches

### ISSUE THREAD

**Issue by MacLemon** [vector-im/riot-web#14496]
### Description
The new room list always resets to default setup when launching the Element/Electron application.

### Steps to reproduce

- Launch Element
- Take some time to change the panel sizes in the room list to show what you actually want to see (within the narrow limits granted)
- Quit Element
- Launch Element again
- All your room list sizes have been reset to the default.

It doesn't remember which panels are expanded and which are collapsed.
It doesn't remember how many items you want to see in a certain panel/category.

Describe how what happens differs from what you expected.
Logs being sent:no, UI/UX issue

I try setting up the room list so it is efficient for me to use.

For example I like my favourited rooms to be all visible at all times. (10 to be exact.) This is where I spend most of my time and where most of my communication takes place.

![New Room List, always resets](https://user-images.githubusercontent.com/90635/87577867-dc05fb00-c6d3-11ea-8e30-1e7eb40e8935.jpg)

The unnecessary “show less” item (with horizontal scroll bar?) just wastes additional vertical space.
(There's already a bunch of issues regarding vertical density…)

When I now quit Element/Desktop and relaunch it again the sidebar is presented like this.
![Screen_Shot_2020-07-15_at_19_41_26](https://user-images.githubusercontent.com/90635/87577821-cabcee80-c6d3-11ea-801b-d6b53974f9cd.jpg)

This now wastes so much vertical space that all my efforts to make the room list usable on a 15" laptop screen (in full height) are gone because Element autoresets them.

Element should always remember the exact room list panel setup and reproduce it exactly upon subsequent launches. Horizontally collapsing panels that cause also the already problematic vertical setup to also collapse cause a very bad user experience.

Basically it boils down to having to setup the room list over and over again, each and every time one launches Element.

### Version information
- **Platform**: Desktop

For the desktop app:
- **OS**: macOS 10.13.6 (High Sierra)
- **Version**: 1.7.0

**Comment by turt2live** [comment]
Duplicate of https://github.com/vector-im/riot-web/issues/14114

**Comment by MacLemon** [comment]
I disagree that these issues are about the same thing, though they sound similar.

#14114 is about having the same setup across multiple clients/sessions as I understand it. For example when using Element on a Desktop and a Laptop and about synching settings between them.

This issue is about having the new room list not reappear as I set it up, across multiple launches of *the same application* and *on the same machine and screen setup*.

Their technical origin may be common across these issues.

**Comment by turt2live** [comment]
@MacLemon can you send logs for when this happens please? I can't reproduce it.

**Comment by MacLemon** [comment]
If you cannot reproduce it, please try with a non-fullscreen window. I'm using a window which is 50% width of my 15.4" laptop screen.

Happened again when relaunching Element directly after updating from 1.7.0 to 1.7.1.

I guess this is Electron specific, not macOS specific and also depends on the window size.


I tried collecting logs two times now, but gave up after waiting 15 minutes each. (Usually this takes but a minute.) I cannot tell if logs were collected successfully and submitted or not. Element 1.7.1 is stuck at a spinner telling me “collecting logs”.

**Comment by lampholder** [comment]
I think this just happened to me on upgrading my electron instance on windows. Relaunched after the upgrade (triggered by accepting the upgrade prompt) and my roomlist came back in what seems to be the default configuration - ordered by activity for people, alphabet for rooms; preview off on both; show with unread first off for both.
